### PR TITLE
Using module path in PS Core

### PR DESCRIPTION
**Pull request by narrieta**
On PSCore, the PowerShell paths are under "...\PowerShell\..." instead of "...\WindowsPowerShell\...".

Added Get-PSModulePathItem to handle those differences.

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/dscresource.tests/154)

### CODE REVIEW

File: TestHelper.psm1 [hunk 1]
```diff
@@ -719,7 +719,7 @@ function Import-PSScriptAnalyzer
     if ($null -eq $psScriptAnalyzerModule)
     {
         Write-Verbose -Message 'Installing PSScriptAnalyzer from the PowerShell Gallery'
-        $psScriptAnalyzerModulePath = "$env:userProfile\Documents\WindowsPowerShell\Modules\PSScriptAnalyzer"
+        $psScriptAnalyzerModulePath = "$(Get-PSModulePathItem -Item UserProfile)\PSScriptAnalyzer"
```
Comment (narrieta) [inline TestHelper.psm1]: done

File: TestHelper.psm1 [hunk 2]
```diff
@@ -743,7 +743,7 @@ function Import-xDscResourceDesigner
     if ($null -eq $xDscResourceDesignerModule)
     {
         Write-Verbose -Message 'Installing xDscResourceDesigner from the PowerShell Gallery'
-        $xDscResourceDesignerModulePath = "$env:userProfile\Documents\WindowsPowerShell\Modules\xDscResourceDesigner"
+        $xDscResourceDesignerModulePath = "$(Get-PSModulePathItem -Item UserProfile)\xDscResourceDesigner"
```
Comment (narrieta) [inline TestHelper.psm1]: done

File: TestHelper.psm1 [hunk 3]
```diff
@@ -888,6 +888,38 @@ function Get-CommandNameParameterValue
     return $nameParameterValue
 }
 
+<#
+    .SYNOPSIS
+        Returns the requested item in $env:PSModulePath
```
Comment (narrieta) [inline TestHelper.psm1]: refactored the function; hopefully the new code is clearer

File: TestHelper.psm1 (hunk 2 above)
Comment (narrieta) [inline TestHelper.psm1]: done

File: TestHelper.psm1 [hunk 4]
```diff
@@ -888,6 +888,38 @@ function Get-CommandNameParameterValue
...
+#>
+function Get-PSModulePathItem {
+    param(
+        [Parameter(Mandatory)]
+        [ValidateSet('PSHome', 'UserProfile')]
+        [string] $Item
+    )
+
+    $itemParent = @{
+        PSHome = $PSHOME
```
Comment (narrieta) [inline TestHelper.psm1]: done

File: TestHelper.psm1 [hunk 5]
```diff
@@ -888,6 +888,38 @@ function Get-CommandNameParameterValue
...
+        UserProfile = $env:USERPROFILE
+    }
+
+    $userProfilePSModulePath = $env:PSModulePath.Split(';') |
```
Comment (narrieta) [inline TestHelper.psm1]: done

File: TestHelper.psm1 [hunk 6]
```diff
@@ -888,6 +888,38 @@ function Get-CommandNameParameterValue
...
+        UserProfile = $env:USERPROFILE
+    }
+
+    $userProfilePSModulePath = $env:PSModulePath.Split(';') | 
+        Where-Object { $_ -like "$($itemParent[$Item])\*" } |
+        Sort-Object -Property Length |
+        Select-Object -First 1
+
+    if (!$userProfilePSModulePath) {
+        Write-Error "Cannot find the requested item in the PowerShell module path.`n`$env:PSModulePath = $env:PSModulePath"
```
Comment (narrieta) [inline TestHelper.psm1]: done

File: TestHelper.psm1 [hunk 7]
```diff
@@ -888,6 +888,38 @@ function Get-CommandNameParameterValue
...
+    }
+
+    $userProfilePSModulePath
```
Comment (narrieta) [inline TestHelper.psm1]: done

File: TestHelper.psm1 [hunk 8]
```diff
@@ -888,6 +888,38 @@ function Get-CommandNameParameterValue
...
+}
+
 Export-ModuleMember -Function @(
     'New-Nuspec', `
```
Comment (narrieta) [inline TestHelper.psm1]: done

File: TestHelper.psm1 [hunk 9]
```diff
@@ -925,6 +927,76 @@ function Get-CommandNameParameterValue
     return $nameParameterValue
 }
 
+<#
+    .SYNOPSIS
+        Returns first the item in $env:PSModulePath that matches the given Prefix ($env:PSModulePath is list of semicolon-separated items).
+        If no items are found, it reports an error.
+    .PARAMS Prefix
```
Comment (kwirkykat) [inline TestHelper.psm1]: .PARAMS --> .PARAMETER

File: Meta.Tests.ps1 [hunk 10]
```diff
@@ -464,18 +464,17 @@ Describe 'Common Tests - Validate Example Files' -Tag 'Examples' {
         <#
             For Appveyor builds copy the module to the system modules directory so it falls in to a PSModulePath folder and is
             picked up correctly.
-            For a user to run the test, they need to make sure that the module exist in one of the paths in env:PSModulePath, i.e.
-            '%USERPROFILE%\Documents\WindowsPowerShell\Modules'.
-            No copying is done when a user runs the test, because that could potentially be destructive.
+            For a user to run the test, they need to make sure that the module exist in one of the paths in env:PSModulePath.
+            No copying is done when a user runs the test, becuase that could potentially be destructive.
```
Comment (narrieta) [inline Meta.Tests.ps1]: done

File: TestHelper.psm1 [hunk 11]
```diff
@@ -925,6 +927,76 @@ function Get-CommandNameParameterValue
...
+            C:\Program Files\WindowsPowerShell\Modules;C:\Users\foo\Documents\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules
+        and the current user is 'foo', then 
+            Get-UserProfilePSModulePathItem 
+        will return 
+            C:\Users\foo\Documents\WindowsPowerShell\Modules
+#>
+function Get-UserProfilePSModulePathItem {
+    param()
+
+    return Get-PSModulePathItem $env:USERPROFILE
```
Comment (kwirkykat) [inline TestHelper.psm1]: Please add the name of the parameter (-Prefix)

File: TestHelper.psm1 [hunk 12]
```diff
@@ -925,6 +927,76 @@ function Get-CommandNameParameterValue
...
+            C:\Program Files\WindowsPowerShell\Modules;C:\Users\foo\Documents\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules
+        then 
+            Get-PSHomePSModulePathItem 
+        will return 
+            C:\Windows\system32\WindowsPowerShell\v1.0\Modules
+#>
+function Get-PSHomePSModulePathItem {
+    param()
+
+    return Get-PSModulePathItem $global:PSHOME
```
Comment (kwirkykat) [inline TestHelper.psm1]: Please add the name of the parameter (-Prefix)

File: TestHelper.psm1 (hunk 9 above)
Comment (narrieta) [inline TestHelper.psm1]: done

File: TestHelper.psm1 (hunk 11 above)
Comment (narrieta) [inline TestHelper.psm1]: done

File: TestHelper.psm1 (hunk 12 above)
Comment (narrieta) [inline TestHelper.psm1]: done

**Pull request merged**